### PR TITLE
Refactor Test Part I

### PR DESCRIPTION
**Pull request by jcsalterego**
Addresses the beginning of #118:

* moves parametrized tests from `test_isos.py` into `class BaseTestISO`
* standardizes on `def test_{method}_*` for ease of use, e.g. `test_get_historical_lmp` becomes `test_get_lmp_historical`
* creates subclasses, e.g. `class TestErcot(BaseTestISO)` in parallel files, e.g. `gridstatus/tests/test_ercot2.py`
* adds a `@with_markets` decorator which wraps `pytest.mark.parametrize("market")`

Once we finalize the approach, we can either:

* merge this PR, possibly creating double testing,
* OR continue porting tests into the subclasses, and then collapsing the parallel files into the final files, e.g. `gridstatus/tests/test_ercot2.py -> gridstatus/tests/test_ercot.py`

### CODE REVIEW

File: gridstatus/tests/test_ercot2.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+from gridstatus import Ercot, NotSupported
+from gridstatus.tests.base_test_iso import BaseTestISO
+
+
+class TestErcot(BaseTestISO):
+    iso = Ercot()
+
+    """get_fuel_mix"""
+
+    @pytest.mark.skip
```
Comment (jcsalterego) [inline gridstatus/tests/test_ercot2.py:12]: This is an extraneous decorator, but I thought it was a good way of noting the ISO is opting-out of a test in addition to the `pass`.

The other options I see are adding a decorator to the BaseTestISO method which includes or excludes certain subclasses, but at the point it seems like six of one, half of a dozen of the other.

File: gridstatus/tests/test_isos.py [hunk 2]
```diff
@@ -1,52 +1,13 @@
...
-    )
-
-
-def check_status(df):
-    assert set(df.columns) == set(
-        ["Time", "Status", "Notes"],
-    )
-
-
 def check_is_datetime_type(series):
```
Comment (kmax12) [inline gridstatus/tests/test_isos.py:11]: i think this was moved to `base_test_iso.py` and is no longer used here

File: gridstatus/tests/test_isos.py [hunk 3]
```diff
@@ -1,52 +1,13 @@
 import pandas as pd
```
Comment (kmax12) [inline gridstatus/tests/test_isos.py:1]: I think we can rename this file now that is just the misc tests that remain. perhaps `test_gridstatus.py`?

perhaps also just leave a comment that ISO general tests go in `base_test_iso` and ISO specific test go in their respective files, so it clear what goes here.

File: gridstatus/tests/test_ercot2.py (hunk 1 above)
Comment (kmax12) [reply gridstatus/tests/test_ercot2.py:12]: I like this approach. i think it's most scalable to have the subclass handle their own opting out. seems better if the base test class doesn't have to be aware of all it's subclasses 

one thought I have to make it even more clear is add a reason to the marker like this `@pytest.mark.skip(reason="Not Applicable")`.

**Review by kmax12 — CHANGES_REQUESTED** [review]
Nice. This must have been some tedious work. Thanks!

I dont see why we can't merge before completing the second part. In fact, the longer this is unmerged, the more git conflicts it'll lead to if new tests are added.

Just a few comments

**Pull request merged**